### DISCUSSION TITLE
foaf purpose

### DISCUSSION THREAD

**Discussion by bogdanaanei** [eclipse-rdf4j/rdf4j#3311] [Q&A] [answered]
Hello,

In documentation, is described the  rdf triple for “Picasso’s first name is Pablo” as **http://example.org/Picasso foaf:firstName "Pablo"**
Why is needed foaf, in front of :firstName?
Thanks

**Comment by barthanssens** [comment, marked as answer]
Hi,

the `foaf:` is a namespace prefix (shorthand) for "http://xmlns.com/foaf/0.1/", so the full predicate (property) becomes http://xmlns.com/foaf/0.1/firstName

As mentioned in the documentation, the idea of using RDF is to clearly define what a property means, and to reuse definitions from existing "vocabularies" like Friend-of-a-Friend, so other people can easily check what is meant by e.g. `firstName`.
You can use other vocabularies, or create your own vocabulary, but for the sake of this example foaf: is a good choice.

It's not mandatory to write `foaf:firstName`, (http://xmlns.com/foaf/0.1/firstName would also work)

**Reply by bogdanaanei** [reply]
Thanks for the answer,

Still, seems clear enough that :firstName refer to Pablo. Can  :firstName be understood to refer something else? It would generate an error if foaf or "http://xmlns.com/foaf/0.1/" is missing?

Bogdan

**Reply by barthanssens** [reply]
In this example it looks indeed quite simple, but in RDF the goal is to always precisely define the properties / relations, to give them  a semantic meaning.
Something like "salary" would be more challenging: is that before or after taxes ? Including special benefits ? Per day/month/year ? Etc
Note that the same RDF data can be saved in various formats (RDF/XML, JSON-LD, ...), so the notation in files may look slightly different from the example.

If "foaf" would not be present, it may indeed produce an error, or a parser might assume (depending on format and settings) it is part of a default namespace / domain set elsewhere, so internally the end result will once again be something like http://example.com/firstName

In the perfect world, there would only be one such property with a good and universally accepted definition, but in reality there are various overlapping vocabularies.
For instance, in the example the property https://schema.org/givenName could also be used (often abbreviated to schema:givenName)
Which one is "best" is up for debate: it may depend on knowledge or use of specific vocabularies, the definitions being used etc.

But it is possible (not required) to somewhat document the relations between properties of different vocabularies in a machine-friendly format.

**Reply by bogdanaanei** [reply]
Thank you very much for your answer. It is all clear now
